src/components/structures/MatrixChat.tsx

```diff
@@ -1132,6 +1133,29 @@ export default class MatrixChat extends React.PureComponent<IProps, IState> {
                 ));
             }
         }
+
+        const client = MatrixClientPeg.get();
+        const plEvent = roomToLeave.currentState.getStateEvents(EventType.RoomPowerLevels, '');
+        const plContent = plEvent ? (plEvent.getContent() || {}) : {};
+        const userLevels = plContent.users || {};
+        const currentUserLevel = userLevels[client.getUserId()];
+        const userLevelValues = Object.values(userLevels);
+        if (isNumberArray(userLevelValues)) {
+            const maxUserLevel = Math.max(...userLevelValues);
+            // If the user is the only user with highest power level
+            if (maxUserLevel === currentUserLevel &&
+                userLevelValues.lastIndexOf(maxUserLevel) == userLevelValues.indexOf(maxUserLevel)) {
+                warnings.push((
+                    <span className="warning" key="last_admin_warning">
+                        { ' '/* Whitespace, otherwise the sentences get smashed together */ }
+                        { _t("You are the sole person with the highest role in this room. " +
+                        "If you leave, the room could become unmoderable. Consider giving " +
+                        "another user your role.") }
```

Comment:
I think we should hide this warning if the "You are the only person here" warning is shown

Reply:
That is already the case. `leaveRoomWarnings(roomId: string)` returns early if the "You are the only person here" warning is added, so no further warnings are shown in that case.